bgpd/bgp_lcommunity.c

```diff
@@ -318,6 +318,12 @@ int lcommunity_cmp(const void *arg1, const void *arg2)
 	const struct lcommunity *lcom1 = arg1;
 	const struct lcommunity *lcom2 = arg2;
 
+	if (lcom1 == NULL && lcom2 == NULL)
```

Comment:
This test for null pointers in a hash has been bothering me, in some places in the code we test and in some places we don't test.  I've been struggling understanding why you would allow a hash_get with a null pointer?  This seems like a broken problem up earlier in the call tree than here.  Thoughts on this?

Reply:
The problem here is that `lcommunity_cmp()` is used on other places other than by the hash code. Please take a look at the `community_list_entry_lookup()` function for example, which is where the reported crash happens.